bblocks/import_tools/ilo.py

```diff
@@ -0,0 +1,212 @@
+
+    for name, link in glossaries_links.items():
+        d.update({name: pd.read_csv(link, index_col=0).iloc[:, 0].to_dict()})
+
+    d.update({"obs_status": d.pop("obs")})
+
+    return d
+
+
+def extract_data(indicator_code: str) -> pd.DataFrame:
+    """Extract data from ILO website
+
+    Args:
+        indicator_code (str): Indicator code to extract
+
+    Returns:
+        pd.DataFrame: Dataframe with the data
+
+    """
+
+    url = (
+        f"{BASE_URL}/ilostat-files/WEB_bulk_download/indicator/{indicator_code}.csv.gz"
+    )
+
+    try:
+        return pd.read_csv(url, compression="gzip")
+    except HTTPError:
+        logger.error(f"Indicator not available: {indicator_code}")
+        raise ValueError(f"Indicator not available: {indicator_code}")
+
+
+def clean_df(df: pd.DataFrame, glossaries: dict[str:dict]) -> pd.DataFrame:
+    """Clean a dataframe
+
+    Args:
+        df (pd.DataFrame): Dataframe to clean
+        glossaries (dict): Dictionary with the names as keys and a dictionary of codes and names as values
+
+    Returns:
+        pd.DataFrame: Cleaned dataframe
+    """
+
+    # loop through glossaries and rename columns
+    for key in glossaries:
+        if key in df.columns:
+            df = df.rename(columns={key: f"{key}_code"})  # rename columns to *_code
+            df[key] = df[f"{key}_code"].map(
+                glossaries[key]
+            )  # add new column with glossary names
+
+    return df
+
+
+def download_data(indicator: str, path: str, glossaries: dict) -> None:
+    """Pipeline to download an indicator and save it to disk.
+
+    Args:
+        indicator: Indicator code to download
+        path: Path to save the data to
+        glossaries: dictionary to map codes to names
+    """
+
+    (extract_data(indicator).pipe(clean_df, glossaries).to_csv(path, index=False))
+
+
+@dataclass
+class ILO(ImportData):
+    """An object to import data from the ILO website
+
+    To use this object, first create an instance of it and then call the `load_data` method
+    to load an indicator to the object. Use the `get_data` method to get the data from the object.
+    To update the data saved on disk and in the object, call the `update_data` method.
+    The attribute `available_indicators` contains a dataframe with information on
+    all the available indicators from the ILO.
+    """
+
+    _glossaries: dict = None  # dictionary used to map codes to names
+    _available_indicators: pd.DataFrame = (
+        None  # dataframe with information on all available indicators
+    )
+
+    @property
+    def available_indicators(self) -> pd.DataFrame:
+        """Return a datafra"""
+
+        if self._available_indicators is None:
+            self._available_indicators = pd.read_csv(
+                f"{BASE_URL}//ilostat-files/WEB_bulk_download/indicator/table_of_contents_en.csv"
+            )
+            logger.info("Loaded available indicators to object")
+        return self._available_indicators
```

Comment:
This should probably not be a property. With properties, it is expected that they will be cheap to compute, so the IDE always runs them on the variable viewer. However, in this case, that means downloading something - which if it fails, means the object creation will run into trouble.

Therefore, I suggest you make this a "get available indicators" so that it is clear that a get operation is involved.

Also given what you are logging is pretty specialised, I would do a logger.debug() instead of info.